internal/services/network/web_application_firewall_policy_resource.go

```diff
@@ -607,6 +636,27 @@ func expandWebApplicationFirewallPolicyWebApplicationFirewallCustomRule(input []
 			RuleType:        webapplicationfirewallpolicies.WebApplicationFirewallRuleType(ruleType),
 		}
 
+		if rateLimitDuration := v["rate_limit_duration"].(string); rateLimitDuration != "" {
+			result.RateLimitDuration = pointer.To(webapplicationfirewallpolicies.ApplicationGatewayFirewallRateLimitDuration(rateLimitDuration))
+		}
+
+		if rateLimitThreshHold := v["rate_limit_threshold"].(int); rateLimitThreshHold > 0 {
+			result.RateLimitThreshold = pointer.To(int64(rateLimitThreshHold))
+		}
+
+		if groupBy := v["group_rate_limit_by"].(string); len(groupBy) > 0 {
```

Comment:
Why are we length checking a string instead of just checking if it's empty like on line 639?